mne/time_frequency/tfr.py

```diff
@@ -331,16 +339,15 @@ def cwt(X, Ws, use_fft=True, mode='same', decim=1):
                                     with Morlet wavelets
     """
     n_signals, n_times = X[:, ::decim].shape
-    n_frequencies = len(Ws)
 
     if use_fft:
-        coefs = _cwt_fft(X, Ws, mode)
+        coefs = _cwt_fft(X, Ws, mode, decim=decim)
     else:
-        coefs = _cwt_convolve(X, Ws, mode)
+        coefs = _cwt_convolve(X, Ws, mode, decim=decim)
 
-    tfrs = np.empty((n_signals, n_frequencies, n_times), dtype=np.complex)
+    tfrs = np.empty((n_signals, len(Ws), n_times), dtype=np.complex)
     for k, tfr in enumerate(coefs):
```

Comment:
If `_cwt_fft()` returns an array of exactly the right size there would be no benefit. Previously, it would have returned an array of the wrong size (too big, undecimated) so there was memory savings in doing it this way. You could change `_cwt_fft` and `_cwt_convolve` to have `return_generator=True` by default and have it be `False` here (meaning, return a filled array).